app/src/androidTest/java/com/epfl/drawyourpath/mainpage/fragments/StatsFragmentTest.kt

```diff
@@ -0,0 +1,811 @@
+            .perform(ViewActions.click())
+
+        // AVERAGE SPEED PER YEAR DISPLAYED
+        checkAverageSpeedYearView()
+
+        Espresso.onView(ViewMatchers.withId(R.id.changeLeftStats))
+            .perform(ViewActions.click())
+
+        // Global stats displayed
+        checkGlobalStatsView()
+
+        scenario.close()
+    }
+
+    /**
+     * Check that the information's displayed (in period month state) when we click on the arrow button on the left are corrected(check the transitions between the fragment views)
+     * The correct transition is Global Stats, number of path goal, total time goal, total distance goal, average distance (with year period), average distance (with month period),
+     * average Duration (with month period), Average Speed(with month period) and global stats.
+     */
+    @Test
+    fun correctTransitionWhenClickingOnLeftArrowButtonWithMonthPeriod() {
+        val scenario =
+            launchFragmentInContainer(themeResId = R.style.Theme_Bootcamp) {
+                mockStatsFragment
+            }
+        // check that the fragment is displayed
+        Espresso.onView(ViewMatchers.withId(R.id.fragment_stats))
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+
+        // Global stats displayed
+        checkGlobalStatsView()
+
+        Espresso.onView(ViewMatchers.withId(R.id.changeLeftStats))
+            .perform(ViewActions.click())
+
+        // NUMBER PATH GOAL DISPLAYED
+        checkPathNumberGoalView()
+
+        Espresso.onView(ViewMatchers.withId(R.id.changeLeftStats))
+            .perform(ViewActions.click())
+
+        // TIME GOAL DISPLAYED
+        checkTimeGoalView()
+
+        Espresso.onView(ViewMatchers.withId(R.id.changeLeftStats))
+            .perform(ViewActions.click())
+
+        // DISTANCE GOAL DISPLAYED
+        checkDistanceGoalView()
+
+        Espresso.onView(ViewMatchers.withId(R.id.changeLeftStats))
+            .perform(ViewActions.click())
+
+        // DISTANCE PER YEAR DISPLAYED
+        checkAverageDistanceYearView()
+
+        Espresso.onView(ViewMatchers.withId(R.id.toggleStats))
+            .perform(ViewActions.click())
+
+        // DISTANCE PER MONTH DISPLAYED
+        checkAverageDistanceMonthView()
+
+        Espresso.onView(ViewMatchers.withId(R.id.changeLeftStats))
+            .perform(ViewActions.click())
+
+        // DURATION PER MONTH DISPLAYED
+        checkAverageDurationMonthView()
+
+        Espresso.onView(ViewMatchers.withId(R.id.changeLeftStats))
+            .perform(ViewActions.click())
+
+        // AVERAGE SPEED PER MONTH DISPLAYED
+        checkAverageSpeedMonthView()
+
+        Espresso.onView(ViewMatchers.withId(R.id.changeLeftStats))
+            .perform(ViewActions.click())
+
+        // Global stats displayed
+        checkGlobalStatsView()
+
+        scenario.close()
+    }
+
+    /**
+     * Test that the transition by clicking n the toggle button are correct in the average speed view.
```

Comment:
Small typo (that is the only thing I can find, all the tests are great)

Reply:
Thank you